src/frontend/js/widgets/Dashboard/components/DashboardItem/Order/OrderStateTeacherMessage/index.tsx

```diff
@@ -0,0 +1,105 @@
+import { FormattedMessage, defineMessages } from 'react-intl';
+import { useEffect } from 'react';
+import {
+  CertificateOrder,
+  CredentialOrder,
+  OrderState,
+  ContractDefinition,
+  NestedCourseOrder,
+} from 'types/Joanie';
+import { StringHelper } from 'utils/StringHelper';
+import { handle } from 'utils/errors/handle';
+import { OrderHelper } from 'utils/OrderHelper';
+
+export const messages = defineMessages({
+  statusDraft: {
+    id: 'components.DashboardItem.Order.OrderStateTeacherMessage.statusDraft',
+    description: 'Status shown on the dashboard order item when order is draft.',
+    defaultMessage: 'Draft',
+  },
+  statusSubmitted: {
+    id: 'components.DashboardItem.Order.OrderStateTeacherMessage.statusSubmitted',
+    description: 'Status shown on the dashboard order item when order is submitted.',
+    defaultMessage: 'Submitted',
+  },
+  statusPending: {
+    id: 'components.DashboardItem.Order.OrderStateTeacherMessage.statusPending',
+    description: 'Status shown on the dashboard order item when order is pending.',
+    defaultMessage: 'Pending',
+  },
+  statusOnGoing: {
+    id: 'components.DashboardItem.Order.OrderStateTeacherMessage.statusOnGoing',
+    description:
+      'Status shown on the dashboard order item when order is validated with no certificate',
+    defaultMessage: 'On going',
+  },
+  statusCompleted: {
+    id: 'components.DashboardItem.Order.OrderStateTeacherMessage.statusCompleted',
+    description:
+      'Status shown on the dashboard order item when order is validated with certificate',
+    defaultMessage: 'Completed',
+  },
+  statusWaitingSignature: {
+    id: 'components.DashboardItem.Order.OrderStateTeacherMessage.statusWaitingSignature',
+    description:
+      "Status shown on the dashboard order item when order is validated with contract's learner signature missing.",
+    defaultMessage: 'Signature required',
+  },
+  statusCanceled: {
+    id: 'components.DashboardItem.Order.OrderStateTeacherMessage.statusCanceled',
+    description: 'Status shown on the dashboard order item when order is canceled',
+    defaultMessage: 'Canceled',
+  },
+  statusOther: {
+    id: 'components.DashboardItem.Order.OrderStateTeacherMessage.statusOther',
+    description: 'Status shown on the dashboard order item when order status is unknown',
+    defaultMessage: '{state}',
+  },
+});
+
+interface OrderStateTeacherMessageProps {
+  order: CredentialOrder | CertificateOrder | NestedCourseOrder;
+  contractDefinition?: ContractDefinition;
+}
+
+// TODO(rlecellier): set the right message for order displayed in the teacher dashbord
+const OrderStateTeacherMessage = ({ order, contractDefinition }: OrderStateTeacherMessageProps) => {
```

Comment:
What do you about using only `OrderStateMessage` but having a `teacherMode: bool` in order to select the messages we want to display? 

The main issue I see having two component is about maintenance and future evolutions, we will have to maintain two mirror component with the same conditional statements and I think this pattern is a kind of a logic duplication 🤔

Reply:
Both would work, i like to split it as a dedicated components.
The role of these components is to give the right message on a given context (learner/teacher).
I think it's easier to read and maintain having two smaller components than a bigger one with conditions.

Reply:
Here is a example with some pseudo code to avoid logic duplication:


```typescript
const messages = {
	
  statusWaitingSignatureStudent: {
	id: 'components.DashboardItem.Order.OrderStateMessage.statusWaitingSignature.student',
    defaultMessage: 'Signature required',
  },

  statusWaitingSignatureTeacher: {
	id: 'components.DashboardItem.Order.OrderStateMessage.statusWaitingSignature.teacher',
    defaultMessage: 'The student must sign',
  },
}

const OrderStateMessage = ({ order, contractDefinition, mode = "student" }: OrderStateMessageProps) => {
  
  const getMessage = (messageId) => {
  	return messages[messageId + "." + mode]
  }


  useEffect(() => {
    if (!Object.values(OrderState).includes(order.state)) {
      handle(new Error(`Unknown order state ${order.state}`));
    }
  }, [order.state]);

  if (order.state === OrderState.VALIDATED) {
    if (OrderHelper.orderNeedsSignature(order, contractDefinition)) {
      return <FormattedMessage {...getMessage("statusWaitingSignature")} />;
    }
  ...
};

Reply:
It's duplicate in purpose.
Like we have a component for Contract status and a other for OrderStates, they're the same but with different context.
I feel like the usage in learner dashboard and teacher dashboard is an enough different context to justify two components.